.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,144 @@
+name: "CodeQL"
+
+on:
+  - push
+  - workflow_dispatch
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        java-version:
+          - 8
+        java-distribution:
+          - adopt
+        language: ['java']
+    steps:
+    - name: Checkout killbill
+      uses: actions/checkout@v2
+      with:
+        repository: killbill/killbill
+        ref: ${{ github.ref }}
+        path: killbill
+        # We must fetch at least the immediate parents so that if this is
+        # a pull request then we can checkout the head.
+        fetch-depth: 2
+    # If this run was triggered by a pull request event, then checkout
+    # the head of the pull request instead of the merge commit.
+    - run: git checkout HEAD^2
+      if: ${{ github.event_name == 'pull_request' }}
+    - name: Setup Java
+      uses: actions/setup-java@v2
+      with:
+        distribution: ${{ matrix.java-distribution }}
+        java-version: ${{ matrix.java-version }}
+    - name: Configure Sonatype mirror
+      uses: s4u/maven-settings-action@v2.3.0
+      # Go to Sonatype directly to avoid delay syncs (could get rid of this if actions/setup-java were to support mirrors).
+      with:
+        mirrors: '[{"id": "oss-releases", "name": "Sonatype releases", "mirrorOf": "*", "url": "https://oss.sonatype.org/content/repositories/releases/"}]'
+    - name: Check if killbill-oss-parent SNAPSHOT must be fetched
```

Comment:
I guess I am confused on all the steps necessary for `SNAPSHOT` - do we need this ?

Reply:
Yes :-) I can fill you in IRL.